recOrder/io/visualization.py

```diff
@@ -0,0 +1,164 @@
+    # Convert 180 degree range into 360 to match periodicity of hue.
+    ori_ = orientation * 360 / np.pi
+    overlay_final = np.zeros_like(retardance)
+
+    # FIX ME: this binning code leads to artifacts.
+    # levels = 32
+    # ori_binned = (
+    #     np.round(orientation[i] / 180 * levels + 0.5) / levels - 1 / levels
+    # ) # bin orientation into 32 levels.
+    # ori_ = np.interp(ori_binned, (0, 1), (0, 360))
+
+    if cmap == "JCh":
+        noise_level = 1
+
+        J = ret_
+        C = np.ones_like(J) * 60
+        C[ret_ < noise_level] = 0
+        h = ori_
+
+        JCh = np.stack((J, C, h), axis=-1)
+        JCh_rgb = cspace_convert(JCh, "JCh", "sRGB1")
+
+        JCh_rgb[JCh_rgb < 0] = 0
+        JCh_rgb[JCh_rgb > 1] = 1
+
+        overlay_final = JCh_rgb
+    elif cmap == "HSV":
+        I_hsv = np.moveaxis(
+            np.stack(
+                [
+                    ori_ / 360,
+                    np.ones_like(ori_),
+                    ret_ / np.max(ret_),
+                ]
+            ),
+            source=0,
+            destination=-1,
+        )
+        overlay_final = hsv_to_rgb(I_hsv)
+    else:
+        raise ValueError(f"Colormap {cmap} not understood")
+
+    return np.moveaxis(
+        overlay_final, source=-1, destination=0
+    )  # .shape = (3, ...)
+
+
+def ret_ori_phase_overlay(
+    czyx, max_val_V: float = 1.0, max_val_S: float = 1.0
+):
+    """
+    Creates an overlay of retardance, orientation, and phase.
+    Maps orientation to hue, retardance to saturation, and phase to value.
+
+    HSV encoding of retardance + orientation + phase image with hsv colormap
+    (orientation in h, retardance in s, phase in v)
+    Parameters
+    ----------
+        czyx        : numpy.ndarray
+                    czyx[0] corresponds to the retardance image
+                    czyx[1]is the orientation image (range from 0 to pi)
+                    czyx[2] is the the phase image
+
+        max_val_V   : float
+                      raise the brightness of the phase channel by 1/max_val_V
+
+        max_val_S   : float
+                      raise the brightness of the retardance channel by 1/max_val_S
+
+    Returns
+    -------
+    overlay                 (nd-array) RGB image with shape (3, ...)
+
+    Returns:
+        RGB with HSV
+    """
+
+    if czyx.shape[0] != 3:
+        raise ValueError(
+            f"Input must have shape (3, ...) instead of ({czyx.shape[0]}, ...)"
+        )
+
+    czyx_out = np.zeros_like(czyx, dtype=np.float32)
+
+    retardance = czyx[0]
+    orientation = czyx[1]
+    phase = czyx[2]
+
+    # Normalize the stack
+    ordered_stack = np.stack(
+        (
+            # Normalize the first channel by dividing by pi
+            orientation / np.pi,
+            # Normalize the second channel and rescale intensity
+            rescale_intensity(
+                retardance,
+                in_range=(
+                    np.min(retardance),
```

Comment:
I originally had set it up to do np.min(retardance), but maybe the right thing to have here is to clip with (0,max(retardance)). What do you think?

Reply:
I think the current behavior is an okay start. I think iterating on good default will require testing across a wide range of datasets. I'd like to do this, but I'd vote to keep the scope of this PR tighter.